man/systemd-nspawn.xml

```diff
@@ -1352,6 +1352,58 @@ After=sys-subsystem-net-devices-ens1.device</programlisting>
         make them read-only, using <option>--bind-ro=</option>.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--bind-user=</option></term>
+
+        <listitem><para>Binds the home directory of the specified user on the host into the container. Takes
+        the name of an existing user on the host as argument. May be used multiple times to bind multiple
+        users into the container. This does three things:</para>
+
+        <orderedlist>
+          <listitem><para>The user's home directory is bind mounted from the host into
+          <filename>/run/hosts/home/</filename>.</para></listitem>
+
+          <listitem><para>An additional UID/GID mapping is added that maps the host user's UID/GID to a
+          container UID/GID, allocated from the 60514…60577 range.</para></listitem>
```

Comment:
Or "allocated from the 60514…60577 range" rather means whatever that range value maps to on the host, i.e. if you have:
```
0:10000:65536
```

then `60518` would be `160518`.

Reply:
> Ok, wrapping my head around this: "allocated from the 60514…60577 range" refers to a range on the host? 

No. Refers to range in the container.

This is about mapping a host user account (e.g. UID 1000) into a container, where we need a suitable UID and pick it from the 60514…60577 range there (e.g. 60517). We then establish a userns mapping mapping host 1000 → container 60517

Reply:
> Ok, wrapping my head around this: "allocated from the 60514…60577 range" refers to a range on the host? So then you create a userns on the host with an idmapping where `1000 -> 60518` and attach it to the user's mount. Let's say `60518 -> 1002`, i.e. `60518` maps to `1002` in the container. Then you create a new user with uid `1002` in the container with your host user's login credentials. Then you login which attaches you to the containers usern's where `1002` maps to `60518` on the host. So if you go through the idmapped mount as uid `1002` in the container your creations will end up as `60518->1000` ending up with uid 1000 on disk.

So this should be:
You create a userns on the host with an idmapping where `1000` is mapped to `160518` and attach the userns to the mount. Then you create an entry for `1000`/"rando" on the host as `60518`/"rando" in the container. When you then log in and create files through the idmapped mount in the host user's home you get `60518 -> 160518 -> 1000` and end up with correct ownership.

Reply:
this stuff doesn't use idmapped mounts. it bind mounts the host home dir into the container as is, but because the container processes map the host 1000 to the container 60517 all will look good and pretty

Reply:
Oh, you're not hotplugging the mount you're setting the container up with that mount.
So you're adding a mapping like:

```
0:10000:60517
60517:1000:1
60518:160518:<the rest>
```

Reply:
Ah yep, seems so:
https://github.com/systemd/systemd/pull/19555/files#diff-3e88b6405bcaf989c1ff90c722b131c15b11bc1c3856c2b1a6fb375f12efafa5R4109

Reply:
yupp